src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -222,9 +235,15 @@ static class OptionsValues {
         boolean ignoreSigning = false;
         boolean bindServices = false;
         boolean suggestProviders = false;
+        boolean ignoreModifiedRuntime = false;
+        boolean generateLinkableRuntime = false;
     }
 
     public static final String OPTIONS_RESOURCE = "jdk/tools/jlink/internal/options";
+    // meta-data files per module for linkable JDK runtimes
+    public static final String RESPATH_PATTERN = "jdk/tools/jlink/internal/runtimelink/fs_%s_files";
+    // The diff files per module for linkable JDK runtimes
+    public static final String DIFF_PATTERN = "jdk/tools/jlink/internal/runtimelink/diff_%s";
```

Comment:
I think it'd be useful to have a `LinkableRuntimeImage` class that declares these constants and some other utility methods such as detecting if this is a linkable runtime image (e.g. `hasRuntimeImageLinkCap()`)